app/bundles/ReportBundle/Entity/Report.php

```diff
@@ -409,4 +467,149 @@ public function getSettings()
+     */
+    public function setToAddress($toAddress)
+    {
+        $this->toAddress = $toAddress;
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getScheduleUnit()
+    {
+        return $this->scheduleUnit;
+    }
+
+    /**
+     * @param null|string $scheduleUnit
+     */
+    public function setScheduleUnit($scheduleUnit)
+    {
+        $this->scheduleUnit = $scheduleUnit;
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getScheduleDay()
+    {
+        return $this->scheduleDay;
+    }
+
+    /**
+     * @param null|string $scheduleDay
+     */
+    public function setScheduleDay($scheduleDay)
+    {
+        $this->scheduleDay = $scheduleDay;
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getScheduleMonthFrequency()
+    {
+        return $this->scheduleMonthFrequency;
+    }
+
+    /**
+     * @param null|string $scheduleMonthFrequency
+     */
+    public function setScheduleMonthFrequency($scheduleMonthFrequency)
+    {
+        $this->scheduleMonthFrequency = $scheduleMonthFrequency;
+    }
+
+    public function setAsNotScheduled()
+    {
+        $this->setIsScheduled(false);
+        $this->setToAddress(null);
+        $this->setScheduleUnit(null);
+        $this->setScheduleDay(null);
+        $this->setScheduleMonthFrequency(null);
+    }
+
+    public function ensureIsDailyScheduled()
+    {
+        $this->setIsScheduled(true);
+        $this->setScheduleUnit(SchedulerEnum::UNIT_DAILY);
+        $this->setScheduleDay(null);
+        $this->setScheduleMonthFrequency(null);
+    }
+
+    /**
+     * @throws ScheduleNotValidException
+     */
+    public function ensureIsMonthlyScheduled()
+    {
+        if (
+            !array_key_exists($this->getScheduleMonthFrequency(), SchedulerEnum::getMonthFrequencyForSelect()) ||
+            !array_key_exists($this->getScheduleDay(), SchedulerEnum::getDayEnumForSelect())
+        ) {
+            throw new ScheduleNotValidException();
+        }
+        $this->setIsScheduled(true);
+        $this->setScheduleUnit(SchedulerEnum::UNIT_MONTHLY);
+    }
+
+    /**
+     * @throws ScheduleNotValidException
+     */
+    public function ensureIsWeeklyScheduled()
+    {
+        if (!array_key_exists($this->getScheduleDay(), SchedulerEnum::getDayEnumForSelect())) {
+            throw new ScheduleNotValidException();
+        }
+        $this->setIsScheduled(true);
+        $this->setScheduleUnit(SchedulerEnum::UNIT_WEEKLY);
+        $this->setScheduleMonthFrequency(null);
+    }
+
+    public function isScheduledDaily()
+    {
+        return $this->getScheduleUnit() === SchedulerEnum::UNIT_DAILY;
+    }
+
+    public function isScheduledWeekly()
+    {
+        return $this->getScheduleUnit() === SchedulerEnum::UNIT_WEEKLY;
+    }
+
+    public function isScheduledMonthly()
+    {
+        return $this->getScheduleUnit() === SchedulerEnum::UNIT_MONTHLY;
+    }
+
+    public function isScheduledWeekDays()
```

Comment:
I don't see this as an option in the "Every" select.

Reply:
Interface has 3 method - `is scheduled daily` / `is scheduled weekly` and `is scheduled monthly`. For weekly and monthly schedule you could select a day or week days, which is a different method.